### PR TITLE
Add edit feature

### PR DESCRIPTION
**Pull request by linnn-7**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/CliSyntax.java [hunk 1]
```diff
@@ -6,11 +6,11 @@
...
-    public static final Prefix PREFIX_EMAIL = new Prefix(":email");
-    public static final Prefix PREFIX_YEAR_JOINED = new Prefix(":year");
-    public static final Prefix PREFIX_ADDRESS = new Prefix("a/");
-    public static final Prefix PREFIX_TAG = new Prefix("t/");
+    public static final Prefix PREFIX_NAME = new Prefix(":name ");
+    public static final Prefix PREFIX_PHONE = new Prefix(":phone ");
+    public static final Prefix PREFIX_EMAIL = new Prefix(":email ");
+    public static final Prefix PREFIX_YEAR_JOINED = new Prefix(":year ");
+    public static final Prefix PREFIX_ADDRESS = new Prefix(":address ");
+    public static final Prefix PREFIX_TAG = new Prefix(":tag ");
```
Comment (PateShin) [inline src/main/java/seedu/address/logic/parser/CliSyntax.java:14]: Are white spaces necessary for Prefixes? 
Perhaps consider to remove the white spaces.
```":name "``` to ```":name"```

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 2]
```diff
@@ -74,12 +74,16 @@ public CommandResult execute(Model model) throws CommandException {
...
+                .filter(person -> person.getId().value == (this.id.value))
+                .findFirst()
+                .orElse(null);
+        if (personToEdit == null) {
+            throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_ID);
         }
-
-        Person personToEdit = lastShownList.get(index.getZeroBased());
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
+        System.out.println(editedPerson);
```
Comment (PateShin) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:86]: Should the code be printed out? 
If not, consider to remove this.

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 3]
```diff
@@ -74,12 +74,16 @@ public CommandResult execute(Model model) throws CommandException {
...
+        Person personToEdit = lastShownList
+                .stream()
+                .filter(person -> person.getId().value == (this.id.value))
+                .findFirst()
+                .orElse(null);
+        if (personToEdit == null) {
+            throw new CommandException(MESSAGE_INVALID_PERSON_DISPLAYED_ID);
         }
-
-        Person personToEdit = lastShownList.get(index.getZeroBased());
```
Comment (PateShin) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:81]: I like the way you rewrote the code for personToEdit and handled any null values with exceptions.

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 4]
```diff
@@ -58,13 +58,15 @@ public EditCommand parse(String args) throws ParseException {
         if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
             editPersonDescriptor.setAddress(ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get()));
         }
-        parseTagsForEdit(argMultimap.getAllValues(PREFIX_TAG)).ifPresent(editPersonDescriptor::setTags);
-
-        if (!editPersonDescriptor.isAnyFieldEdited()) {
-            throw new ParseException(EditCommand.MESSAGE_NOT_EDITED);
+        if (parseTagsForEdit(argMultimap.getAllValues(PREFIX_TAG)).isEmpty()) {
+            System.out.println(!editPersonDescriptor.isAnyFieldEdited());
```
Comment (PateShin) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:62]: Similar to **line 86** in EditCommand.java.
Maybe consider to remove the code?

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 5]
```diff
@@ -48,6 +50,8 @@ public PersonCard(Person person, int displayedIndex) {
         super(FXML);
         this.person = person;
         id.setText(displayedIndex + ". ");
+        System.out.println(person.getId().value);
```
Comment (PateShin) [inline src/main/java/seedu/address/ui/PersonCard.java:53]: Again same thing happen here as discussed above:

Is print needed here? If not, consider to remove.

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 6]
```diff
@@ -65,4 +65,23 @@ public static boolean isNonZeroUnsignedInteger(String s) {
...
+     * e.g. empty string, "-1", "0", "+1", and " 2 " (untrimmed), "3 0" (contains whitespace), "1 a" (contains letters)
+     * @throws NullPointerException if {@code s} is null.
+     */
+    public static boolean isSixDigitNumber(String s) {
+        requireNonNull(s);
+
+        try {
+            int value = Integer.parseInt(s);
+            return value >= 100000
+                    && value <= 999999 && !s.startsWith("+"); // "+1" is successfully parsed by Integer#parseInt(String)
```
Comment (PateShin) [inline src/main/java/seedu/address/commons/util/StringUtil.java:82]: Have you considered whether there might be a more optimal approach to writing this section of code? 
For example: `return s.matches("[1-9]\\d{5}");`

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 7]
```diff
@@ -74,7 +88,7 @@ public static Phone parsePhone(String phone) throws ParseException {
      */
     public static Address parseAddress(String address) throws ParseException {
         requireNonNull(address);
-        String trimmedAddress = address.trim();
+        String trimmedAddress = address.replace(" :tag", "").trim();
```
Comment (PateShin) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:91]: Is `.replace(" :tag", "")` needed?

**Review by PateShin — COMMENTED** [review]
Overall, the modified codes for **edit feature** clear to read👍

There are just a few minor issues and suggestions that could be improved.

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 8]
```diff
@@ -65,4 +65,23 @@ public static boolean isNonZeroUnsignedInteger(String s) {
...
+     * Will return false for any other non-null string input
+     * e.g. empty string, "-1", "0", "+1", and " 2 " (untrimmed), "3 0" (contains whitespace), "1 a" (contains letters)
+     * @throws NullPointerException if {@code s} is null.
+     */
+    public static boolean isSixDigitNumber(String s) {
+        requireNonNull(s);
+
+        try {
+            int value = Integer.parseInt(s);
+            return value >= 100000
```
[the remark this answers is not in the document]
Comment (linnn-7) [reply src/main/java/seedu/address/commons/util/StringUtil.java:81]: Thanks for your suggestion! Changed to return s.matches("[1-9]\d{5}").

File: src/main/java/seedu/address/logic/parser/CliSyntax.java (hunk 1 above)
Comment (linnn-7) [reply src/main/java/seedu/address/logic/parser/CliSyntax.java:14]: fixed

**Pull request merged**